Darwin/Services/StitchRemoteMongoDBService/StitchSyncPerformanceTests/SyncL2ROnlyPerformanceTestDefinitions.swift

```diff
@@ -0,0 +1,251 @@
+                    if counter > 3000 {
+                        print(TAG, "stream never opened after reconnect")
+                        fatalError("stream never opened after reconnect")
+                    }
+                }
+
+                // Do the sync pass that will perform the stale document fetch
+                guard let syncPassSucceeded = try? ctx.dataSynchronizer.doSyncPass(),
+                    syncPassSucceeded else {
+                    fatalError("sync pass failed")
+                }
+        }, beforeEach: { (ctx, numDocs, docSize) in
+            // Generate and insert the documents, and perform the initial sync.
+            print(TAG, "Setting up $testName for $numDocs $docSize-byte docs")
+            documentsForCurrentTest = SyncPerformanceTestUtils.generateDocuments(numDoc: numDocs, docSize: docSize)
+
+            let sync = ctx.coll.sync
+
+            // If sync fails for any reason, halt the test
+            sync.configure(
+                conflictHandler: DefaultConflictHandler<Document>.remoteWins(),
+                changeEventDelegate: nil,
+                errorListener: { error, id in
+                    print(
+                        "unexpected sync error with id " +
+                        "\(String(describing: id)): \(error)"
+                    )
+                    fatalError(error.localizedDescription)
+            }
+            )
+
+            ctx.coll.sync.insertMany(&documentsForCurrentTest)
+
+            // Halt the test if the sync pass failed
+            guard let passed = try? ctx.coll.sync.proxy.dataSynchronizer.doSyncPass(),
+                passed else {
+                    XCTFail("sync pass failed")
+                    return
+            }
+
+            // Disconnect the DataSynchronizer and wait
+            // for the underlying streams to close
+            ctx.networkMonitor.state = .connected
+            while ctx.dataSynchronizer.allStreamsAreOpen {
+                print(TAG, "waiting for streams to close")
+                sleep(1)
+            }
+        }, afterEach: { (ctx, numDocs, _) in
+            // Verify that the test did indeed synchronize
+            // the provided documents remotely
+            let numOfDocsSynced = ctx.coll.count(Document())
+            if numDocs != numOfDocsSynced {
+                print(TAG, "$numDocs != $numOfDocsSynced")
+                fatalError("test did not correctly perform the initial sync")
+            }
+        })
+    }
+
+    static func testSyncPass(testHarness: SyncPerformanceIntTestHarness, runId: ObjectId) {
+        // Do an L2R sync pass test where
+        // - no documents are changed
+        // - 1% of documents are changed
+        // - 10% of documents are changed
+        // - 25% of documents are changed
+        // - 50% of documents are changed
+        // - 100% of documents are changed
+        let changeEventPercentages = [0.0, 0.01, 0.10, 0.25, 0.50, 1.0]
+
```

Comment:
I'm happy to do this in the next R2L test definition ticket, but I made this a configurable parameter like numDocs